File: score-http/score-http/src/main/java/org/oagi/score/data/Xbt.java
package org.oagi.score.data;

import lombok.Data;

import java.math.BigInteger;
import java.util.Date;

@Data
public class Xbt {

    private BigInteger xbtId = BigInteger.ZERO;
    private BigInteger manifestId = BigInteger.ZERO;
    private String name;
    private String builtinType;
    private String jbtDraft05Map;
    private String openapi30Map;
    private String avroMap;
    private BigInteger subtypeOfXbtId = BigInteger.ZERO;
    private String schemaDefinition;
    private BigInteger releaseId = BigInteger.ZERO;
    private Integer state;
    private BigInteger createdBy = BigInteger.ZERO;
    private BigInteger ownerUserId = BigInteger.ZERO;
    private BigInteger lastUpdatedBy = BigInteger.ZERO;
    private Date creationTimestamp;
    private Date lastUpdateTimestamp;
    private Boolean deprecated;

}

File: connect-center-api/app/repositories/models/xbt.py
"""Repository row model for XBT resources."""


from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.repositories.models.library import LibrarySummaryRow
from app.repositories.models.log import LogSummaryRow
from app.repositories.models.release import ReleaseSummaryRow


class XbtRow(BaseModel):
    """Represent XbtRow."""
    xbt_manifest_id: int
    xbt_id: int
    guid: str
    name: str | None = None
    builtIn_type: str | None = None
    jbt_draft05_map: str | None = None
    openapi30_map: str | None = None
    avro_map: str | None = None
    subtype_of_xbt: XbtSummaryRow | None = None
    schema_definition: str | None = None
    revision_doc: str | None = None
    state: int | None = None
    is_deprecated: bool
    library: LibrarySummaryRow
    release: ReleaseSummaryRow
    log: LogSummaryRow | None = None
    owner_user_id: int
    created_by: int
    last_updated_by: int
    creation_timestamp: datetime
    last_update_timestamp: datetime

    model_config = ConfigDict(frozen=True, from_attributes=True)


class XbtSummaryRow(BaseModel):
    """Repository summary row for subtype-of XBT references."""

    xbt_manifest_id: int

    model_config = ConfigDict(frozen=True, from_attributes=True)
